libs-scala/scala-utils/src/main/scala/com/daml/scalautil/NeverEqualsOverride.scala

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.scalautil
+
+/** This trait is useful to be mixed in if:
+  * - equality semantics for the base type is not an issue
+  * - class has non-comparable typed fields
+  *
+  * Practical usage: mixing in in DTO ADT top trait, which has case class implementors having Array fields: disables equality semantics, and prevents Wartremover warnings.
```

Comment:
For arrays specifically, I would use `SomeArrayEquals` for this (this can be promoted from the Speedy library to `scala-utils`), along with its tests.

Reply:
I would completely agree if the purpose would be to support equality of arrays. Here the purpose is different: we explicitly do not care about equality (hence the name)

Reply:
Isn't this rather dangerous for future usage? What if someone comes and starts implementing tests wrongly relying on the case class contract?

Reply:
well, the test would fail I suppose? :) or the same what would happen if you would not have wartremover and would want to compare two case classes with fields without equality on the contract...
This guy is more about the wartremover warnings than case classes/contracts in general.
Personally I would rather go for this relaxation, so we can reap the benefit and cut the noise, but happy to change if you see fit.